### PR TITLE
feature/unify sql store

### PR DESCRIPTION
**Pull request by gfyrag**
- Refactor storage to reduce duplicate code between different sql engines.
- Add some utils for testing

### CODE REVIEW

File: storage/sqlstorage/driver.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/numary/ledger/storage"
+)
+
+type Flavor = sqlbuilder.Flavor
+
+var (
+	SQLite     = sqlbuilder.SQLite
+	PostgreSQL = sqlbuilder.PostgreSQL
+	MySQL      = sqlbuilder.MySQL
```
Comment (altitude) [inline storage/sqlstorage/driver.go]: I think we can remove MySQL until we have a migration file that works and do the whole thing (i.e. run tests on MySQL as well)

File: storage/sqlstorage/driver.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		driverName: "sqlite3",
+	},
+	PostgreSQL: {
+		driverName: "pgx",
+	},
+}
+
+type ConnStringResolver func(name string) string
+
+// cachedDBDriver is a driver which connect on the database each time the NewStore() method is called.
```
Comment (altitude) [inline storage/sqlstorage/driver.go]: openCloseDbDriver probably

**Review by altitude — APPROVED** [review]

File: storage/sqlstorage/driver.go (hunk 2 above)
Comment (gfyrag) [reply storage/sqlstorage/driver.go]: Yes

File: storage/sqlstorage/driver.go (hunk 1 above)
Comment (gfyrag) [reply storage/sqlstorage/driver.go]: Yes good idea.

**Pull request merged**